File: frontend/src/types/invites.ts
import type { InviteTokenStatus } from "@/types/enums";

export interface InviteTokenCreate {
  email: string;
}

export interface InviteTokenRead {
  id: number;
  token_hash: string;
  email: string;
  company_name: string | null;
  contact_first_name: string | null;
  contact_last_name: string | null;
  note: string | null;
  status: InviteTokenStatus;
  created_by_admin_id: number;
  created_at: string;
  expires_at: string;
  used_at: string | null;
}

export interface InviteMetadataPublic {
  email: string;
}

File: src/schemas/invites.py
"""Invite token schemas."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict, EmailStr, Field

from src.enums import InviteTokenStatus


class InviteTokenCreate(BaseModel):
    """Admin input for creating an invite — email only."""

    email: EmailStr = Field(..., max_length=255)


class InviteTokenRead(BaseModel):
    """Full invite record returned to admin."""

    model_config = ConfigDict(from_attributes=True)

    id: int
    token_hash: str
    email: str
    company_name: str | None
    contact_first_name: str | None
    contact_last_name: str | None
    note: str | None
    status: InviteTokenStatus
    created_by_admin_id: int
    created_at: datetime
    expires_at: datetime
    used_at: datetime | None


class InviteMetadataPublic(BaseModel):
    """Safe pre-fill data returned to the registration page."""

    model_config = ConfigDict(from_attributes=True)

    email: str
